src/mrpro/operators/functionals/l1.py

```diff
@@ -0,0 +1,72 @@
+"""L1 Norm."""
+
+import torch
+
+from mrpro.operators._Functional import ProximableFunctional
+
+
+class L1Norm(ProximableFunctional):
+    """Functional for L1 Norm.
+
+    Parameters
+    ----------
+        lambda = 1
+    """
+
+    def forward(self, x: torch.Tensor) -> tuple[torch.Tensor]:
+        """forward.
+
+        Args:
+            x (torch.Tensor): data tensor
+
+        Returns
+        -------
+            tuple[torch.Tensor]: forward of data
+        """
+        return (torch.tensor([(x - self.g).abs().sum(self.dim)]),)
+
+    def prox(self, x: torch.Tensor, sigma: torch.Tensor) -> tuple[torch.Tensor]:
+        """Prox of L1 Norm.
+
+        Args:
+            x (torch.Tensor): data tensor
+            sigma (torch.Tensor): scaling factor
+
+        Returns
+        -------
+            tuple[torch.Tensor]: prox of data
+        """
+        is_complex = x.is_complex()
+        if is_complex:
+            x = torch.view_as_real(x)
+            threshold = torch.tensor([self.lam * sigma]).unsqueeze(-1)
+        else:
+            threshold = torch.tensor([self.lam * sigma])
+        x = torch.clamp(x, -threshold, threshold)
```

Comment:
x = torch.nn.functional.relu(x - threshold) - torch.nn.functional.relu(- x - threshold)

Reply:
there should be a g somewhere in there

Reply:
i think there is no g since here we conly compute the soft-threshold of the input x, i.e. prox_{threshold * || ||_1}, the g, i.e. usually the data, will be mostly present only when we have the l2-norm.

Reply:
there should be a g.

Reply:
also, for complex inputs, this should be modrelu, i.e.:

`polar(relu(abs(x)-thres), angle(x))`

Reply:
for complex, this should be modrelu, i.e. something like
```
diff=x-g
thres=lambda*sigma
prox=polar(relu(abs(d)-thres), angle(d))
```